### ISSUE TITLE
storage.NewMergeSeriesSet expects series sets to be ordered

### ISSUE THREAD

**Issue by pstibrany** [prometheus/prometheus#6717]
## Proposal
`storage.NewMergeSeriesSet` expects series sets to be ordered, but this requirement isn't documented. If sets are not ordered, merging them fails to produce expected result.

I suggest to add a mention of this requirement to documentation for NewMergeSeriesSet.

Here is a test case (can be added to `TestMergeSeriesSet`) which shows invalid result with sets which have series in different order.

```
{
	input: []SeriesSet{newMockSeriesSet(
		newMockSeries(labels.FromStrings("bar", "baz"), []sample{{1, 1}, {2, 2}}),
		newMockSeries(labels.FromStrings("foo", "bar"), []sample{{0, 0}, {1, 1}}),
	), newMockSeriesSet(
		newMockSeries(labels.FromStrings("foo", "bar"), []sample{{2, 2}, {3, 3}}),
		newMockSeries(labels.FromStrings("bar", "baz"), []sample{{3, 3}, {4, 4}}),
	)},
	expected: newMockSeriesSet(
		newMockSeries(labels.FromStrings("bar", "baz"), []sample{{1, 1}, {2, 2}, {3, 3}, {4, 4}}),
		newMockSeries(labels.FromStrings("foo", "bar"), []sample{{0, 0}, {1, 1}, {2, 2}, {3, 3}}),
	),
},
```

* Prometheus version:

master, 9adad8ad3045a93fff03400970f1aa77cc034cb0.

**Comment by bwplotka** [comment]
Valid request, help wanted. That's why also Thanos gRPC StoreAPI.Series() has to return sorted series as well.

**Comment by codesome** [comment]
Closed by #6718
